### ISSUE TITLE
Week #9 goals

### ISSUE THREAD

**Issue by zengchen221** [qlcchain/go-qlc#148]
- Ledger @zengchen221 
    - #135 Add a table for storage blocks relationship 
    - #143 refactor block struct 
    - #145 add ledger interface to get block by phone number
   
- VM @gythialy 
    - #139 implement Pack/Unpack contract abi
    - #140 implement smart contract on chain

- P2P @pwc5232759 
    - #147 add doc about UT and IT for p2p module and consensus module
    - #146 add UT and IT for p2p module and consensus module
    - #144 optimize log printing for p2p modules and consensus modules 
    - #123 Synchronized blocks do not require consensus 
    - #124 Circulation of consensus messages

- Timestamp @wenyuangui 
    - #142 Draw a solution about the timestamp of the chain

- Consensus @ariesunny 
   - #128 Summary a doc about DPOS consensus

**Comment by zengchen221** [comment]
#140 blocked by #131 , #145, #146, #147  postponed to next week , #142 need review
